src/plugins/mavlink_ftp/include/plugins/mavlink_ftp/mavlink_ftp.h

```diff
@@ -0,0 +1,189 @@
+
+    /**
+     * @brief Callback type for `list_directory_async()` call to get directory items and result.
+     */
+    typedef std::function<void(Result, std::vector<std::string>)>
+        directory_items_and_result_callback_t;
+
+    /**
+     * @brief Callback type for `calc_file_crc32_async()` call to get file CRC32.
+     */
+    typedef std::function<void(Result, uint32_t)>
+        file_crc32_result_callback_t;
+
+    /**
+     * @brief Downloads a file to local folder (asynchronous).
+     *
+     * @param remote_file_path Remote file to download
+     * @param local_folder Local folder where downloaded file will be stored
+     * @param callback Callback to receive result of this request.
+     */
+    void download_async(std::string remote_file_path,
+                        std::string local_folder,
+                        progress_callback_t progress_callback,
+                        result_callback_t result_callback);
+
+    /**
+     * @brief Uploads local file to remote folder (asynchronous).
+     *
+     * @param local_file_path Local file to upload
+     * @param remote_folder Remote folder where uploaded file will be stored
+     * @param callback Callback to receive result of this request.
+     */
+    void upload_async(std::string local_file_path,
+                      std::string remote_folder,
+                      progress_callback_t progress_callback,
+                      result_callback_t result_callback);
+
+    /**
+     * @brief Downloads a vector of directory items from the system (asynchronous).
+     *
+     * @param path Path of a directory to list
+     * @param callback Callback to receive directory entries and result of this request.
+     */
+    void list_directory_async(std::string path, directory_items_and_result_callback_t callback);
+
+    /**
+     * @brief Creates directory (asynchronous).
+     *
+     * @param path Path of a directory to create
+     * @param callback Callback to receive result of this request.
+     */
+    void create_directory_async(std::string path, result_callback_t callback);
+
+    /**
+     * @brief Removes directory (asynchronous).
+     *
+     * @param path Path of a directory to remove
+     * @param callback Callback to receive result of this request.
+     */
+    void remove_directory_async(std::string path, result_callback_t callback);
+
+    /**
+     * @brief Removes file (asynchronous).
+     *
+     * @param path Path of a file to remove
+     * @param callback Callback to receive result of this request.
+     */
+    void remove_file_async(std::string path, result_callback_t callback);
+
+    /**
+     * @brief Truncates file (asynchronous).
+     *
+     * @param path Path of a file to remove
+     * @param offset Offset to which file is truncated
+     * @param callback Callback to receive result of this request.
+     */
+    void truncate_file_async(std::string path, uint32_t offset, result_callback_t callback);
```

Comment:
What do we need this for?

Reply:
It is weird, but it is defined in Mavlink FTP spec: https://github.com/mavlink/mavlink-devguide/blob/master/ko/services/ftp.md#TruncateFile
Maybe they wanted to use it to reset a log file?

Reply:
But does a MAVSDK user need it? If not I would not expose it. What we expose we need to test.

Reply:
I agree, this function doesn't seem very useful. For all the others that are not tested yet I will write integration tests.